v2.1/online-schema-changes.md

```diff
@@ -0,0 +1,205 @@
+
+## Limitations
+
+Schema changes cannot be performed inside user [transactions][txns] in the general case. Instead, they run as background jobs (that can be viewed with [`SHOW JOBS`][show-jobs]). This behavior is necessary because making schema changes transactional would mean requiring a given schema change transaction to propagate across all the nodes of a cluster. This would block all user-initiated transactions being run by your application, since the schema-change-containing transaction would have to commit before any other transactions could make progress. This would prevent the cluster from servicing reads and writes during the schema change, requiring application downtime.
+
+For example, this means that the following queries will result in errors.
+
+Create an index and then run a select against that index inside a user-level transaction:
+
+{% include copy-clipboard.html %}
+~~~ sql
+CREATE TABLE foo (id INT PRIMARY KEY, name VARCHAR);
+BEGIN;
+SAVEPOINT cockroach_restart;
+CREATE INDEX foo_idx ON foo (id, name);
+SELECT * from foo_idx;
+RELEASE SAVEPOINT cockroach_restart;
+COMMIT;
+~~~
+
+~~~
+CREATE TABLE
+BEGIN
+SAVEPOINT
+CREATE INDEX
+ERROR:  relation "foo_idx" does not exist
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
+ROLLBACK
+~~~
+
+Add a column and then add a constraint against that column inside a user-level transaction:
+
+{% include copy-clipboard.html %}
+~~~ sql
+CREATE TABLE foo ();
+BEGIN;
+SAVEPOINT cockroach_restart;
+ALTER TABLE foo ADD COLUMN bar VARCHAR;
+ALTER TABLE foo ADD CONSTRAINT bar CHECK (foo IN ('a', 'b', 'c', 'd'));
+RELEASE SAVEPOINT cockroach_restart;
+COMMIT;
+~~~
+
+~~~
+CREATE TABLE
+BEGIN
+SAVEPOINT
+ALTER TABLE
+ERROR:  column "foo" not found for constraint "foo"
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
+ROLLBACK
+~~~
+
+Add a column and then select against that column inside a user-level transaction:
+
+{% include copy-clipboard.html %}
+~~~ sql
+CREATE TABLE foo ();
+BEGIN;
+SAVEPOINT cockroach_restart;
+ALTER TABLE foo ADD COLUMN bar VARCHAR;
+SELECT bar FROM foo;
+RELEASE SAVEPOINT cockroach_restart;
+COMMIT;
+~~~
+
+~~~
+CREATE TABLE
+BEGIN
+SAVEPOINT
+ALTER TABLE
+ERROR:  column name "bar" not found
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
+ROLLBACK
+~~~
+
+{{site.data.alerts.callout_success}}
+As of version 2.1, you *can* run schema changes inside the same transaction as a [`CREATE TABLE`][create-table] statement. For more information, [see this example](#run-schema-changes-inside-a-transaction-with-create-table).
+{{site.data.alerts.end}}
+
+## Examples
+
+### Run schema changes inside a transaction with `CREATE TABLE`
+
+As noted above, and in the list of [known limitations](known-limitations.html#schema-changes-within-transactions), you cannot run schema changes inside user transactions in general.
```

Comment:
If we unify the known limitations entry and the limitations section on this page, you can remove `and in the list...`.

Reply:
Fixed by (1) unifying this page's *Limitations* section using includes, and (2) changing the language to say 

```
As noted in [Limitations](#limitations), you cannot run schema changes inside transactions in general.  However ...
```